tests/runner.py

```diff
@@ -1488,26 +1488,43 @@ def reftest(self, expected, manually_trigger=False):
       }
 ''' % (reporting.read(), basename, int(manually_trigger)))
 
-  def compile_btest(self, args, reporting=True):
+  def compile_btest(self, args, reporting=True, native_reporting=True):
```

Comment:
the name of this variable doesn't seem clear enough to me. is "native" using `REPORT_RESULT` (which is "native" for the browser approach) or using exit codes (which is "native" for code in general)?

could it be `use_report_result`?

Reply:
We have two different options that control the way we report results here:

`reporting=True`:  Include JS helpers for reporting results that can be used by JS code.    This include the onExit hook.   The reason we have this option at all is that there is at least one tests that wants to disable all of this and to the XHR directly itself.  

`native_reporting=True`: In addition to the JS helpers also inject the REPORT_RESULT macro at compile time and include the implementation .cpp file. 

Maybe if I called the first one `js_reporting` it might make `native_reporting` more obvious it its meaning?  Or I could define an enum to make it super clear?

For tests that use btest_exit so want to skip that later (keeps the build simpler) but we still want the JS reporting hooks.

Reply:
`js_reporting` does seem clearer, yes. how about `c_reporting` for the second? even clearer I think.

Reply:
done